### PR TITLE
Edwards25519 basic arithmetic mod 2^255-19

### PR DESCRIPTION
**Pull request by ewynx**
This branch contains basic arithmetic for GF(2^255 - 19). We store elements in radix-51 representation. 
We've used 2 reference implementations:
- NaCl impl (current reference from authors)
https://cr.yp.to/ecdh.html#use
- Go impl (uses same radix-51 rep)
https://cs.opensource.google/go/go/+/master:src/crypto/internal/edwards25519/field/fe.go

The functions in the `field_element` file are all tested, except for `inverse` which is too heavy and gives the Immediate18TooLarge error. 
Also functionality from `ge25519` gives that same error.

### CODE REVIEW

File: testing/tests_bls12_381/tests/utils/mod.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+use fuels::{
```
Comment (Mikerah) [inline testing/tests_bls12_381/tests/utils/mod.rs:1]: Should be cherrypicked into its own branched or removed

**Review by Mikerah — CHANGES_REQUESTED** [review]

**Comment by Mikerah** [comment]
For the most part everything looks good

**Comment by ewynx** [comment]
Cherrypicked the edwards stuff and merged [here](https://github.com/hashcloak/fuel-crypto/pull/6). Separate [commit](https://github.com/hashcloak/fuel-crypto/commit/2232627e3dfd9bd0d05382ab9b435e056993f06a) for the final small fixes. 

Everything that was in this PR is now merged and the tests of bls and edwards run as they did in the PRs.

**Comment by Mikerah** [comment]
Then this PR should be closed

**Pull request closed without merging**